util/otbn_build.py

```diff
@@ -41,52 +41,62 @@
 import subprocess
 import sys
 from pathlib import Path
-from typing import List, Optional
-
-log.basicConfig(level=log.INFO, format="%(message)s")
+from typing import Any, List, Optional
 
 REPO_TOP = Path(__file__).parent.parent.resolve()
 
-def cmd_to_str(cmd):
-    return ' '.join([shlex.quote(str(a)) for a in cmd])
+
+def cmd_to_str(cmd: List[str]) -> str:
+    return ' '.join([shlex.quote(a) for a in cmd])
 
 
-def run_cmd(cmd, **kwargs):
+def run_cmd(cmd: List[str], **kwargs: Any) -> None:
```

Comment:
Requiring explicit str casts to cmd is a step in the wrong direction, IMO, and brings us back to fragile Python 3.5 code.

Reply:
Would you prefer `cmd` to have type `List[object]`? The type conveys much less information, but it does allow us to avoid the casts.

Reply:
You can also make it an union between str and os.PathLike

Reply:
Yeah, I thought that was a bit ugly, but let's try that. I'll push up a version that does that.

Reply:
Hmm, I've just pushed that up. I'm not a fan: lots of explicit types needed to hold MyPy's hand (because it doesn't automatically infer the sum type in the way you'd hope). Shall I just drop the first patch and/or delete the type annotations in the code?